logstash-core/lib/logstash/instrument/metric.rb

```diff
@@ -45,20 +57,14 @@ def time(key, &block)
       end
     end
 
-    # TODO
-    def self.create(name, collector = LogStash::Instrument::Collector.instance)
+    def self.create(namespace, collector = LogStash::Instrument::Collector.instance)
       reporter = LogStash::Instrument::Reporter::Stdout.new(collector)
-      Metric.new(collector, name)
+      Metric.new(collector, namespace)
     end
 
     private
-    def merge_keys(key)
-      valid_key!(key)
-      [@base_key, key.to_sym]
-    end
-    
-    def valid_key!(key)
-      raise MetricNoKeyProvided if key.nil? || key == ""
+    def validate_key!(key)
+      raise MetricNoKeyProvided if key.nil? || key.empty?
```

Comment:
I think it should be done at the data structure level to add this information and not on the library level what do you think?